src/backend/distributed/transaction/backend_data.c

```diff
@@ -720,9 +721,41 @@ AssignDistributedTransactionId(void)
 
 	MyBackendData->transactionId.initiatorNodeIdentifier = localGroupId;
 	MyBackendData->transactionId.transactionOriginator = true;
-	MyBackendData->transactionId.transactionNumber =
-		nextTransactionNumber;
+	MyBackendData->transactionId.transactionNumber = nextTransactionNumber;
 	MyBackendData->transactionId.timestamp = currentTimestamp;
+	MyBackendData->citusInitiatedBackend = true;
+
+	SpinLockRelease(&MyBackendData->mutex);
+
+	MarkCitusInitiatedBackend();
+}
+
+
+/*
+ * MarkCitusInitiatedBackend set that backend is initiated by Citus.
+ */
+void
+MarkCitusInitiatedBackend(void)
+{
+	SpinLockAcquire(&MyBackendData->mutex);
+
+	MyBackendData->citusInitiatedBackend = true;
+	MyBackendData->transactionId.transactionOriginator = true;
```

Comment:
I think setting `MyBackendData->transactionId.transactionOriginator ` is wrong and dangerous. 

At this point, we don't know whether the `transactionId` is valid or not.

Also, why do we need to do that?